csrc/sycl/kernels.cpp

```diff
@@ -1020,7 +1020,7 @@ SYCL_EXTERNAL void kDequantizeBlockwise(float *code, unsigned char * buff_A, flo
     /*
     DPCT1064:96: Migrated __ldg call is used in a macro/template definition and may not be valid for all macro/template uses. Adjust the code.
     */
-    local_abs_max = absmax[(i+item_ct1.get_local_id(2)*NUM_PER_TH)/(blocksize)];
+    local_abs_max =  sycl::ext::oneapi::experimental::cuda::ldg(&absmax[(i+item_ct1.get_local_id(2)*NUM_PER_TH)/(blocksize)]);
```

Comment:
Is this implemented for backend other than CUDA? I can't really find anything in the docs about it other than how this basically translates to the [ld.global.nc PTX instruction](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-ld-global-nc).

Reply:
Yes it is portable across all backends as described in the spec document: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_cuda_tex_cache_read.asciidoc#backend-support-status
However it is a no-op on other backends: it simply returns the value, and does not use any special hardware. For Nvidia it forces usage of the read-only texture cache that can considerably improve performance for relevant use cases.

For information on the purpose/performance of it see https://codeplay.com/portal/blogs/2023/09/11/nvidia-texture-cache-in-sycl